src/assignments/PAUL_OMAKI/assignment07.clj

```
@@ -1 +1,214 @@
+   (-> (client/get (format "%s/%s/" url id-num)
+                   {:as :json
+                    :query-params {:api_token api-key}})
+       (:body))))
+
+
+;; (defn fetch-person
+;;   "Fetches a person's data that matches supplied key and value."
+;;   [key value]
+;;   (fetch-data persons-url key value))
+;; (defn fetch-org
+;;   "Fetches an organization's data that matches supplied key and value."
+;;   [key value]
+;;   (fetch-data organizations-url key value))
+;; (defn fetch-lead
+;;   "Fetches a lead's data that matches supplied key and value."
+;;   [key value]
+;;   (fetch-data leads-url key value))
+
+
+
+(defn fetch-person-by-id
+  "Fetches a person's data that matches supplied ID."
+  [id-num]
+  (get (fetch-data persons-url id-num) :data))
+(defn fetch-org-by-id
+  "Fetches an organization's data that matches supplied ID."
+  [id-num]
+  (get (fetch-data organizations-url id-num) :data))
+(defn fetch-lead-by-id
+  "Fetches a lead's data that matches supplied ID."
+  [id-num]
+  (get (fetch-data leads-url id-num) :data))
+
+(defn fetch-all-persons
+  "Fetches all people in the database."
+  []
+  (get (fetch-data persons-url) :data))
+(defn fetch-all-orgs
+  "Fetches all organizations in the database."
+  []
+  (get (fetch-data organizations-url) :data))
+(defn fetch-all-leads
+  "Fetches all leads in the database."
+  []
+  (get (fetch-data leads-url) :data))
+
+
+;;; Sending new data
+
+(defn post-data
+  "Generic upload for data. Applies the hardcoded API token."
+  [url params-map]
+  (let [response (client/post url
+                              {:as :json
+                               :throw-exceptions false
+                               :query-params {:api_token api-key}
+                               :form-params  (m/map-keys csk/->snake_case_keyword params-map)})]
+    (if (= 200 (:status response))
+      (:body response)
+      {:error "Failed to transfer data." :details (:body response)})))
```

you should throw instead of returning a map